provider/provider.go

```diff
@@ -0,0 +1,24 @@
+package provider
```

Comment:
either your file name and your struct name doesn't show any information what this file is for